### ISSUE TITLE
Bad performance on my own dataset

### ISSUE THREAD

**Issue by Georgegal00** [lucastabelini/PolyLaneNet#67]
Hello , thank you for making this work available. I am currently trying to train the network on my own dataset consisting of images collected in the carla simulator.I have collected about 15,000 total images for train, validation and test set.I am using this configuration file:
`# Training settings
seed: 0
exps_dir: 'experiments'
iter_log_interval: 100
iter_time_window: 100
model_save_interval: 10
backup:
model:
  name: PolyRegression
  parameters:
    num_outputs: 28 # (4 lanes) * (1 conf + 2 (upper & lower) + 4 poly coeffs)
    pretrained: true
    backbone: 'efficientnet-b1'
    pred_category: false
    curriculum_steps: [0, 0, 0, 0]
loss_parameters:
  conf_weight: 1
  lower_weight: 1
  upper_weight: 1
  cls_weight: 0
  poly_weight: 150 #300
batch_size:  16 
epochs: 385
optimizer:
  name: Adam
  parameters:
    lr: 1.0e-3
lr_scheduler:
  name: CosineAnnealingLR
  parameters:
    T_max: 385

# Testing settings
test_parameters:
  conf_threshold: 0.5

# Dataset settings
datasets:
  train:
    type: LaneDataset
    parameters:
      dataset: carla
      split: train
      img_size: [360, 640]
      normalize: true
      aug_chance: 0.9090909090909091 # 10/11
      augmentations:
       - name: Affine
         parameters:
           rotate: !!python/tuple [-10, 10]
       - name: HorizontalFlip
         parameters:
           p: 0.5
       - name: CropToFixedSize
         parameters:
           width: 1152
           height: 648
      root: "/data/dataset/GG"
  test: &test
    type: LaneDataset
    parameters:
      dataset: carla
      split: test
      max_lanes: 4
      img_size: [360, 640]
      root: "/data/dataset/GG"
      normalize: true
      augmentations: []

  val: &val
    type: LaneDataset
    parameters:
      dataset: carla
      split: val
      max_lanes: 4
      img_size: [360, 640]
      root: "/data/dataset/GG"
      normalize: true
      augmentations: []`
The annotations are as in tuismple format, so I am processing them with the same methods
. Running python -m lib.datasets.lane_dataset , the loaded lanes are correct, I attach some pictures.
![image_002](https://github.com/user-attachments/assets/45ab3822-27e0-4841-ba3c-4835f265acc5)
![image_005](https://github.com/user-attachments/assets/4793f038-7f21-4225-a0ad-ea1882458a38)
![image_010](https://github.com/user-attachments/assets/3f8edbfd-54b9-413d-b851-357ff1a8fd1d)
![image_011](https://github.com/user-attachments/assets/798c56c1-6be3-4b88-ad5d-b0374fcac108)
![image_012](https://github.com/user-attachments/assets/723b3671-19fc-4f57-a4ca-ef789a0a8b27)
![image_016](https://github.com/user-attachments/assets/58a19f72-9ab7-470d-9272-8f49d4675655)
![image_017](https://github.com/user-attachments/assets/c8bc0a32-748a-432d-993e-125682b9e20c)
![image_019](https://github.com/user-attachments/assets/77e6b0df-b897-4f42-b84b-8e254def7cc4)
I am experiencing problems in the convergence of the model, the loss of train slowly decreases , while the loss of validation fluctuates between 1.2 and 3 , while the accuracy fluctuates between 0.35 and 0.55, this is at epoch 50 of the train. Analyzing the various logs, it can be seen that the polynomial loss is the predominant one. I also visualized some predictions to see where it was actually going wrong and I noticed that in some images it is getting close, while for others it is off completely, I attach some pictures.
![pred7](https://github.com/user-attachments/assets/daad31e1-77c5-4f12-a006-02b3b9f3578d)
![pred6](https://github.com/user-attachments/assets/85a44805-a08f-49f3-81b1-30261d97899d)
![pred5](https://github.com/user-attachments/assets/b2e1200c-93a8-4b71-bb4b-10f4fe4e72e5)
 I would like to understand if it is a data problem, dataset too small? Or is some adjustment to the model needed? Or do I simply need to figure out what the best parameters are? Thank you very much for your attention, any suggestions or answers are welcome!
